Update README.md (repo structure, features) (#36)
alckasoc/discussion-agents: README.md

# Discussion Agents

## Features

 Our primary goal is to provide easy-to-use and clean functions for natural language processing tasks, enabling developers to build robust and efficient applications with ease.

- Easy-to-Use Interface: Provides intuitive and user-friendly functions for rapid prototyping and development.

- Clean Functions: Offers clean and well-structured functions, promoting readability and maintainability of code.

- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.

## Getting Started 

```python
question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring?'

llm = ChatOpenAI(openai_api_key="YOUR_API_KEY")
agent = ReActAgent(llm=llm)
out = agent.generate(question=question)

```

> Here's what the output would look like!```

Reviewer: ```suggestion
Here's what the output would look like!
```

```
Thought : I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.
Action : Search[best kick boxer in the world]
Observation : (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.
Thought : It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
Action : Lookup[crimes]
Observation: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
Thought: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.
...

> 

Reviewer: ```suggestion
```

# Discusion Agents

<badges>

## Table of Contents

## Features

Welcome to our machine learning library leveraging the power of LLM (Large Language Model). Our primary goal is to provide easy-to-use and clean functions for natural language processing tasks, enabling developers to build robust and efficient applications with ease.

- LLM Integration: Leverages the advanced language processing capabilities of LLM for enhanced accuracy and efficiency.

- Easy-to-Use Interface: Provides intuitive and user-friendly functions for rapid prototyping and development.

- Clean Functions: Offers clean and well-structured functions, promoting readability and maintainability of code.

- Scalability: Designed for scalability, enabling processing of large datasets and complex models with ease.

- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.

> # The library offers:

Reviewer: Merge features, and the library offers.

- Modularized implementations of popular LLM-based agents and agent-related methods.
- A flexible framework to build your own agents using innovations from the literature.

#### Getting started example

```python
question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring'

llm = OpenAI(openai_api_key="YOUR_API_KEY")

> agent = ReActAgent(llm=llm)

Reviewer: not `OpenAI`. The `llm` parameter takes a `BaseChatModel`.

out = agent.generate(question=question)

```

> Output

Reviewer: Shouldn't include full output.

Author: its not the full output

Author: should i shorten the output more

Reviewer: yep

----

**Question** :

I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.

**Action:**: Search[best kick boxer in the world]

**Observation**: (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.

**Thought**: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.

**Action**: Lookup[crimes]

**Observation**: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.

**Thought**: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.

**Action**: Lookup[controversies]

**Observation**: (Result 1/1) Dekkers was known for his aggressive style and has been involved in a number of controversies, including a bar brawl and an altercation with a bouncer.

**Thought**: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about the controversies and crimes.

**Action**: Lookup[unsportsmanlike conducts]

...

---

> The iterative process of thought, action, and observation offers an intuitive and transparent approach to problem-solving.

Reviewer: delete

> ## Developer Setup: Windows

Reviewer: ```suggestion
## Developer Setup
```

> # Discusion Agents

Reviewer: ```suggestion
# Discussion Agents
```

# Discussion Agents

## Features

 Our primary goal is to provide easy-to-use and clean functions for natural language processing tasks, enabling developers to build robust and efficient applications with ease.

- Easy-to-Use Interface: Provides intuitive and user-friendly functions for rapid prototyping and development.

- Clean Functions: Offers clean and well-structured functions, promoting readability and maintainability of code.

- Scalability: Designed for scalability, enabling processing of large datasets and complex models with ease.

- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.

> #### Getting started example

Reviewer: ## Getting Started

```python
question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring?'

> 
> 

Reviewer: ```suggestion

```

llm = ChatOpenAI(openai_api_key="YOUR_API_KEY")
agent = ReActAgent(llm=llm)
out = agent.generate(question=question)

```

Output
----

**Question** : Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring ? 

**Thought** : I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.

**Action**: Search[best kick boxer in the world]
**Observation**: (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.
**Thought**: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
**Action**: Lookup[crimes]
**Observation**: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
**Thought**: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.

> **...**

Reviewer: ```
Thought: I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.
Action: Search[best kick boxer in the world]
Observation: (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.
Thought: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
Action: Lookup[crimes]
Observation: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
Thought: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.
...
```

# Discussion Agents

## Features

 Our primary goal is to provide easy-to-use and clean functions for natural language processing tasks, enabling developers to build robust and efficient applications with ease.

- Easy-to-Use Interface: Provides intuitive and user-friendly functions for rapid prototyping and development.

- Clean Functions: Offers clean and well-structured functions, promoting readability and maintainability of code.

> - Scalability: Designed for scalability, enabling processing of large datasets and complex models with ease.

Reviewer: ```suggestion
```

- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.

> #### Getting started 

Reviewer: ```suggestion
## Getting Started 
```

```python
question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring?'

llm = ChatOpenAI(openai_api_key="YOUR_API_KEY")
agent = ReActAgent(llm=llm)
out = agent.generate(question=question)

```

> Output

Reviewer: ```suggestion
```

- Easy-to-Use Interface: Provides intuitive and user-friendly functions for rapid prototyping and development.

- Clean Functions: Offers clean and well-structured functions, promoting readability and maintainability of code.

- Scalability: Designed for scalability, enabling processing of large datasets and complex models with ease.

- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.

#### Getting started example

```python
question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring?'

llm = ChatOpenAI(openai_api_key="YOUR_API_KEY")
agent = ReActAgent(llm=llm)
out = agent.generate(question=question)

```

Output
----

**Question** : I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.

> 

Reviewer: missing a thought?

> **Question** : I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.
> 
> 
> **Action**: Search[best kick boxer in the world]
> 
> **Observation**: (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.
> 
> **Thought**: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
> 
> **Action**: Lookup[crimes]
> 
> **Observation**: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
> 
> **Thought**: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.
> 
> ...

Reviewer: ```suggestion
**Question** : I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.

**Action**: Search[best kick boxer in the world]
**Observation**: (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.
**Thought**: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
**Action**: Lookup[crimes]
**Observation**: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
**Thought**: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.
**...**
```

# Discussion Agents

## Features

Welcome to our machine learning library leveraging the power of LLM (Large Language Model). Our primary goal is to provide easy-to-use and clean functions for natural language processing tasks, enabling developers to build robust and efficient applications with ease.

> - LLM Integration: Leverages the advanced language processing capabilities of LLM for enhanced accuracy and efficiency.

Reviewer: ```suggestion
```

- Easy-to-Use Interface: Provides intuitive and user-friendly functions for rapid prototyping and development.

- Clean Functions: Offers clean and well-structured functions, promoting readability and maintainability of code.

- Scalability: Designed for scalability, enabling processing of large datasets and complex models with ease.

- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.

> ## The library offers:
> 
> 
> - Modularized implementations of popular LLM-based agents and agent-related methods.
> - A flexible framework to build your own agents using innovations from the literature.

Reviewer: ```suggestion
```

#### Getting started example

```python

> question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring'

Reviewer: ```suggestion
question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring?'
```

llm = ChatOpenAI(openai_api_key="YOUR_API_KEY")
agent = ReActAgent(llm=llm)
out = agent.generate(question=question)

```

Output
----

**Question** :

> I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.

Reviewer: incorrect formatting

Author: so do you want me to do it as same as the Acts and Observataions

poetry install
```

Project Organization
------------

 

    ├── data

>     │   ├── external                        <- Data from third party sources.

Reviewer: Need to make sure all arrows line up

    │   ├── interim                         <- Intermediate data that has been transformed.
    │   ├── processed                       <- The final, canonical data sets for modeling.
    │   └── raw                             <- The original, immutable data dump.
    │
    ├── discussion_agents                   <- Source code for this project.
    │   ├── cog   
    │   │   ├── agent                       <- Model/agent-related modules.
    │   │   │   
    │   │   ├── eval                        <- Agent core modules.
    │   │   │   
    │   │   ├── functional                  
    │   │   │
    │   │   ├── modules           

>     │   │   │   ├── memory      <- Memory-related modules.
>     │   │   │   ├── plan        <- Planning-related modules.
>     │   │   │   ├── reflect      <- Reflecting-related modules.
>     │   │   │   └── score        <- Scoring-related modules.

Reviewer: ```suggestion
    │   │   │   ├── memory      <- Memory-related modules.
    │   │   │   ├── plan            <- Planning-related modules.
    │   │   │   ├── reflect         <- Reflecting-related modules.
    │   │   │   └── score          <- Scoring-related modules.
```

    │   │   │
    │   │   ├── persona             
    │   │   │
    │   │   └── prompts             
    │   │

>     │   └── utils           <- Utility methods.
>     │
>     ├── docs                 <- A default Sphinx project; see sphinx-doc.org for details.

Reviewer: ```suggestion
    │   └── utils             <- Utility methods.
    │
    ├── docs                 <- A default Sphinx project; see sphinx-doc.org for details.
```

poetry install
```

Project Organization
------------

 

    ├── data
    │   ├── external                        <- Data from third party sources.
    │   ├── interim                         <- Intermediate data that has been transformed.
    │   ├── processed                       <- The final, canonical data sets for modeling.
    │   └── raw                             <- The original, immutable data dump.
    │
    ├── discussion_agents                   <- Source code for this project.
    │   ├── cog   
    │   │   ├── agent                       <- Model/agent-related modules.
    │   │   │   ├── base.py                 <-

>     │   │   │   ├── generative_agents.py

Reviewer: don't need to specify every single file within a module. not scalable

Author: then i will just have the folders
